dev/com.ibm.ws.security.mp.jwt_fat/fat/src/com/ibm/ws/security/mp/jwt/fat/CommonMpJwtFat.java

```diff
@@ -0,0 +1,538 @@
+            Log.info(thisClass, "goodTestExpectations", "Failed building expectations: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    public Expectations addClaimExpectations(JwtTokenTools jwtTokenTools, String testAction, String theClass) throws Exception {
+        try {
+            Expectations expectations = new Expectations();
+            if (!theClass.contains("ClaimInjection") || (theClass.contains("ClaimInjection") && theClass.contains("RequestScoped"))) {
+                expectations.addExpectation(addApiOutputExpectation(testAction, "getRawToken", MpJwtFatConstants.MP_JWT_TOKEN, null, jwtTokenTools.getJwtTokenString()));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getIssuer", MpJwtFatConstants.JWT_BUILDER_ISSUER, PayloadConstants.ISSUER));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getSubject", MpJwtFatConstants.JWT_BUILDER_SUBJECT, PayloadConstants.SUBJECT));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getTokenID", MpJwtFatConstants.JWT_BUILDER_JWTID, PayloadConstants.JWTID));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getExpirationTime", MpJwtFatConstants.JWT_BUILDER_EXPIRATION,
+                                                                     PayloadConstants.EXPIRATION_TIME_IN_SECS));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getIssuedAtTime", MpJwtFatConstants.JWT_BUILDER_ISSUED_AT,
+                                                                     PayloadConstants.ISSUED_AT_TIME_IN_SECS));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getAudience", MpJwtFatConstants.JWT_BUILDER_AUDIENCE, PayloadConstants.AUDIENCE));
+                expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getGroups", MpJwtFatConstants.PAYLOAD_GROUPS, "groups"));
+                // we won't have a list of claims to check for ClaimInjection, we don't use the api to retrieve the claims and there is no injected claim that lists all claims...
+                if (!theClass.contains("ClaimInjection")) {
+                    for (String key : jwtTokenTools.getClaims()) {
+                        expectations.addExpectations(addApiOutputExpectation(testAction, jwtTokenTools, "getClaim", MpJwtFatConstants.JWT_BUILDER_CLAIM, key));
+                    }
+                }
+            }
+
+            return expectations;
+        } catch (Exception e) {
+            Log.info(thisClass, "goodTestExpectations", "Failed building expectations: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    public Expectations addApiOutputExpectation(String testAction, JwtTokenTools jwtTokenTools, String api, String keyword, String key) throws Exception {
```

Comment:
Is `api` the name of the API method invoked by the app that we're adding an expectation for here?